File: Dart/Bicycle.dart
import 'Dart:math' as math;

class Wheel {
  double rim, tire;

  Wheel(rim, tire) {
    this.rim = rim;
    this.tire = tire;
  }

  double diameter() {
    return this.rim + (this.tire * 2);
  }

  double circumference() {
    return this.diameter() * math.pi;
  }
}

class Gear {
  double chainring, cog;
  var wheel;

  Gear(chainring, cog, [wheel]) {
    this.chainring = chainring;
    this.cog = cog;
    this.wheel = wheel;
  }

  double ratio() {
    return this.chainring / this.cog;
  }

  double gearInches() {
    return this.ratio() * this.wheel.diameter();
  }
}

void main() {
  Wheel wheel = new Wheel(26.0, 1.5);
  print(wheel.circumference());
  print(Gear(52.0, 11.0).ratio());
  print(Gear(52.0, 11.0, wheel).gearInches());
}

File: Python/Bicycle.py
import math


class Wheel:
    def __init__(self, rim, tire):
        self.__rim = rim
        self.__tire = tire

    @property
    def rim(self):
        return self.__rim

    @property
    def tire(self):
        return self.__tire

    def diameter(self):
        return self.rim + (self.tire * 2)

    def circumference(self):
        return self.diameter() * math.pi


class Gear:
    def __init__(self, chainring, cog, wheel=None):
        self.__chainring = chainring
        self.__cog = cog
        self.__wheel = wheel

    @property
    def chainring(self):
        return self.__chainring

    @property
    def cog(self):
        return self.__cog

    @property
    def wheel(self):
        return self.__wheel

    def ratio(self):
        return self.chainring / float(self.cog)

    def gear_inches(self):
        return self.ratio() * self.wheel.diameter()


if __name__ == "__main__":
    wheel = Wheel(26, 1.5)
    print(wheel.circumference())
    print(Gear(52, 11).ratio())
    print(Gear(52, 11, wheel).gear_inches())
